rest-server/auth-server/src/main/scala/com/actionml/authserver/routes/AuthenticationDirectives.scala

```diff
@@ -25,11 +25,11 @@ import com.actionml.authserver.Realms
 import scala.concurrent.Future
 
 
-trait ClientAuthentication extends Directives {
+trait AuthenticationDirectives extends Directives {
 
-  def authenticateClient(fn: (String, String) => Future[_]): Directive1[String] = extractCredentials.flatMap {
-    case Some(BasicHttpCredentials(clientId, password)) =>
-      onSuccess(fn(clientId, password)).map(_ => clientId)
+  def basicAuth(fn: (String, String) => Future[_]): Directive1[String] = extractCredentials.flatMap {
```

Comment:
What do you mean by `basicAuth`? Are we using OAuth2 or not? If we are using OAuth2 what would you call the type, there are several types described in the OAuth2 spec.

Reply:
oauth2 uses basic auth at some stages, it's absolutely corresponds to spec